pandas/core/frame.py

```diff
@@ -9582,7 +9582,7 @@ def join(
 
         Parameters
         ----------
-        other : DataFrame, Series, or list of DataFrame
+        other : DataFrame, Series, or list of either of these
```

Comment:
I think this should say "or list of DataFrames, Series, or both".  "Either of these" implies that it has to be all one type.